### PR TITLE
fix: repair package structure

### LINKED ISSUE #59: Linting does check all projects anymore
Weirdly, `npm run lint` at top-level doesn't seem like it's looking correctly through each project. This breaks our CI.

### LINKED ISSUE #84: Make sure node_modules are scoped to their respective package
Can be done alonside #59

Self-explanatory

### PR DESCRIPTION
**Pull request by ExiledNarwal28**
### Description

To hell npm workspaces, we do not want backend and frontend to share the same dependencies. This will now run lint correctly and only install needed dependencies in our projects.

### Issues

Closes #84 
Closes #59 

### General PR check list

🚨 Please review the guidelines for contributing (`CONTRIBUTING.md`) to this repository.

- [x] Pass CI (build, tests, lint, ...)
~~- [ ] Add tests for new domain logic~~
- [x] Update documentation (README + `/docs`) (remember that changes to the file structure must be explained in `/docs`)
- [x] Make sure PR author is listed in contributors
- [x] Keep PRs small, otherwise split into chunks (I tried, but this is an architectural refactor :/ )
- [x] Find a way to share a comment `.eslintrc.js` or create a new issue

Thank you for contributing!

### Results

Linting runs correctly, project dependencies are scoped.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-      ]
-    },
-    rules: {
-      'import/extensions': ['off'],
-      '@typescript-eslint/prefer-function-type': ['warn'],
-    },
-    ignorePatterns: [
-      '.eslintrc.js',
-    ],
+ root: true,
```
Comment (ExiledNarwal28) [inline .eslintrc.js:2]: 1. This file will need to be reworked, I changed `parserOptions` and `ignorePatterns` in project-specific `.eslintrc.js`
2. My goal is to make all project's `.eslintrc.js` extend this one, so we don't rewrite the same logic over and over

File: .github/workflows/build-backend-infra.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Build Backend Infra
```
Comment (ExiledNarwal28) [inline .github/workflows/build-backend-infra.yml:1]: No workflows were triggered in this PR. This is weird, I'll look into it.

File: .eslintrc.js (hunk 1 above)
Comment (ExiledNarwal28) [reply .eslintrc.js]: Got it

File: .github/workflows/build-backend-infra.yml (hunk 2 above)
Comment (ExiledNarwal28) [reply .github/workflows/build-backend-infra.yml:1]: OK ÇA MARCHE

**Review by elabonte-at-strat — APPROVED** [review]

**Pull request merged**